graphql_compiler/query_formatting/common.py

```diff
@@ -34,91 +35,84 @@
 ######
 
 
-def _raise_invalid_type_error(name, expected_python_type_name, value):
+def _raise_invalid_type_error(
+    name: str, expected_python_types: Collection[Type], value: Any
+) -> NoReturn:
     """Raise a GraphQLInvalidArgumentError that states that the argument type is invalid."""
     raise GraphQLInvalidArgumentError(
-        "Invalid type for argument {}. Expected {}. Got value {} of "
-        "type {}.".format(name, expected_python_type_name, value, type(value).__name__)
+        f"Invalid type for argument {name}. Expected one of {expected_python_types}. Got value "
+        f"{value} of type {type(value).__name__} instead."
     )
 
 
-def _deserialize_anonymous_json_argument(expected_type: GraphQLScalarType, value: Any) -> Any:
-    """Deserialize argument. See docstring of deserialize_json_argument.
+_CUSTOM_SCALAR_DESERIALIZATION_FUNCTIONS = MappingProxyType(
```

Comment:
Can you add a comment here explaining why this is necessary and what makes these "custom"? There's nothing obviously custom about "parse ints as ints, and floats as floats", so it must be something non-obvious and worth explaining.